karman-avionics/src/utils/FlashMem.c

```diff
@@ -14,66 +14,73 @@
 
 #include <string.h>
 
+/** Address for the first data entry. 0 + sizeof header */
 #define INITIAL_DATA_ADDR (0x00000010L)
 
-
-/* This variable is truly global */
+/** Flash memory control data */
 flashmem_ctrl_t gFlashmemCtrl;
 
-/* Initialize the flash memory module */
+/**
+ * @brief Initialize the flash memory module 
+ */
 void init_flashmem(void)
 {
-    /* Setup default number of entries */
+    /** Setup default number of entries */
     gFlashmemCtrl.data_addr = INITIAL_DATA_ADDR;
     gFlashmemCtrl.header.entry_size = sizeof(flash_data_entry_t);
     gFlashmemCtrl.header.magic = MAGIC_NUMBER;
     gFlashmemCtrl.header.num_entries = 0;
     sprintf(gFlashmemCtrl.header.version_str, VERSION_STRING);
 
-    /* Initialize flash memory driver */
+    /** Initialize flash memory driver */
     init_extflash();
 
-    /* Parse flash memory header */
+    /** Parse flash memory header */
     flashmem_hdrstatus_t headerStatus;
     flash_data_hdr_t header;
 
     headerStatus = flashmem_verify_header(&header);
 
-    /* try once again on read failure */
+    /** try once again on read failure */
     if(headerStatus == HDR_READFAIL)
     {
         headerStatus = flashmem_verify_header(&header);
-        /* If we failed again, just write it off as a loss */
+        /** If we failed again, just write it off as a loss */
         if(headerStatus == HDR_READFAIL)
         {
             headerStatus = HDR_INVALID;
         }
     }
 
-    /* If the header is invalid, write the one we formed at the top of the function */
+    /** If the header is invalid, write the one we formed at the top of the function */
     if(headerStatus != HDR_VALID)
     {
-        /* Try to write the default header twice */
+        /** Try to write the default header twice */
         flashmem_write_header();
         
-        /* If what we read isn't what we wrote, try again... */
+        /** If what we read isn't what we wrote, try again... */
         if(HDR_VALID != flashmem_verify_header(&header))
         {
-            /* If this fails we're kinda screwed as far as flash memory goes. */
+            /** If this fails we're kinda screwed as far as flash memory goes. */
             flashmem_write_header();
         }
     }
     else
     {
-        /* Use the value we found already on the flash memory to inform where to write data */
-        /* This means that any new data will be written following the pre-exisiting entries. */
-        /* In order to ignore the previous entries, the size of entries or the version string */
-        /* must be different from what the code expects. Or the magic number is not correct. */
+        /* Use the value we found already on the flash memory to inform where to write data
```

Comment:
It looks like this is just describing this line so it probably doesn't need to be a doxygen formatted comment but it looks like you are formatting similar comments above as doxygen comments so I wanted to have you double check it.
Needs a second *

Reply:
Added *